source/plugins/ruby/MdmMetricsGenerator.rb

```diff
@@ -276,14 +307,27 @@ def getContainerResourceUtilMetricRecords(recordTimeStamp, metricName, percentag
           thresholdPercentageDimValue: thresholdPercentage,
         }
         records.push(Yajl::Parser.parse(StringIO.new(resourceUtilRecord)))
+
+        # Adding another metric for threshold violation
+        resourceThresholdViolatedRecord = MdmAlertTemplates::Container_resource_threshold_violation_template % {
+          timestamp: recordTimeStamp,
+          metricName: @@container_metric_name_metric_threshold_violated_hash[metricName],
+          containerNameDimValue: containerName,
+          podNameDimValue: podName,
+          controllerNameDimValue: controllerName,
+          namespaceDimValue: podNamespace,
+          containerResourceThresholdViolated: isZeroFill ? 0 : 1,
```

Comment:
Do these metrics get manufactured only when threshold is violated? Thats why you are zero filling them to generate 'violated' metric ?

Reply:
Yes. That is right